pufferlib/ocean/trash_pickup/trash_pickup.h

```diff
@@ -0,0 +1,596 @@
+void render(Client* client, CTrashPickupEnv* env) {
+    if (IsKeyDown(KEY_ESCAPE)) {
+        CloseWindow();
+        free(client);
+        exit(0);
+    }
+
+    BeginDrawing();
+    ClearBackground(RAYWHITE);
+
+    // Draw header with current step and total episode reward
+    int total_trash_not_collected = 0;
+    for (int i = 0; i < env->num_trash; i++){
+        total_trash_not_collected += env->trash_presence[i];
+    }
+    DrawText(
+        TextFormat(
+            "Step: %d\nTotal Episode Reward: %.2f\nTrash Collected: %d/%d",
+            env->current_step,
+            env->total_episode_reward,
+            env->num_trash - total_trash_not_collected,
+            env->num_trash
+        ),
+        5, 2, 10, BLACK
+    );
+
+    // Draw the grid and its elements
+    for (int x = 0; x < env->grid_size; x++) {
+        for (int y = 0; y < env->grid_size; y++) {
+            int cell_type =  env->grid[INDEX(env, x, y)].type;
+            int screen_x = x * client->cell_size;
+            int screen_y = y * client->cell_size + client->header_offset;
+
+            Rectangle cell_rect = {
+                .x = screen_x,
+                .y = screen_y,
+                .width = client->cell_size,
+                .height = client->cell_size
+            };
+
+            // Draw grid cell border
+            DrawRectangleLines((int)cell_rect.x, (int)cell_rect.y, (int)cell_rect.width, (int)cell_rect.height, LIGHTGRAY);
+
+            // Draw grid cell content
+            if (cell_type == EMPTY) {
+                // Do nothing for empty cells
+            } else if (cell_type == TRASH) {
+                DrawRectangle(
+                    screen_x + client->cell_size / 4,
+                    screen_y + client->cell_size / 4,
+                    client->cell_size / 2,
+                    client->cell_size / 2,
+                    BROWN
+                );
+            } else if (cell_type == TRASH_BIN) {
+                DrawRectangle(
+                    screen_x + client->cell_size / 8,
+                    screen_y + client->cell_size / 8,
+                    3 * client->cell_size / 4,
+                    3 * client->cell_size / 4,
+                    BLUE
+                );
+            } else if (cell_type == AGENT) {
+                DrawTexturePro(
+                    client->agent_texture, 
+                    (Rectangle) {0, 0, 128, 128},
+                    (Rectangle) {
+                        screen_x + client->cell_size / 2, 
+                        screen_y + client->cell_size / 2,
+                        client->cell_size,
+                        client->cell_size
+                        },
+                    (Vector2){client->cell_size / 2, client->cell_size / 2},
+                    0,
+                    WHITE
+                );
+
+                // Display if the agent is carrying trash
+                for (int agent_idx = 0; agent_idx < env->num_agents; agent_idx++) {
```

Comment:
Why is this loop inside of the agent conditional? I think it draws every agent every loop